src/pyodide/internal/asgi.py

```diff
@@ -217,3 +210,16 @@ async def fetch(app, req, env):
 
 async def websocket(app, req):
     return await process_websocket(app, req)
+
+
+def __getattr__(name):
+    if name == "env":
+        from fastapi import Depends, Request
+
+        @Depends
+        async def env(request: Request):
+            return request.scope["env"]
```

Comment:
I guess we should move this into the sdk @dom96...

Reply:
Hmm, what is the purpose of this? Isn't it fastapi-specific?

Reply:
Yes.

Reply:
Sequestering it in the `__getattr__` function makes it so that as long as nobody tries to access `env`, it's possible to use asgi without importing fastapi.